frb_codegen/src/config.rs

```diff
@@ -1,70 +1,181 @@
-    #[structopt(long)]
+    #[clap(long, multiple_values = true)]
     pub dart_root: Option<Vec<String>>,
-    /// Skip running build_runner even when codegen-capable code is detected
-    #[structopt(long)]
+    /// Skip running build_runner even when codegen-required code is detected
+    #[clap(long)]
     pub no_build_runner: bool,
     /// Show debug messages.
-    #[structopt(short, long)]
+    #[clap(short, long)]
     pub verbose: bool,
+    /// Enable WASM module generation.
+    /// Requires: --dart-decl-output
+    #[clap(long, requires = "dart-decl-output")]
+    pub wasm: bool,
+}
+
+/// Generic accumulator over the targets.
+///
+/// [`Acc<Option<String>>`] implements [`From<T>`] where <code>T: [ToString]</code>
+/// for code shared between all platforms.
+#[derive(Debug, Default, Clone)]
+pub struct Acc<T> {
+    pub common: T,
+    pub io: T,
+    pub wasm: T,
+}
+
+impl<T> std::ops::AddAssign for Acc<Vec<T>> {
+    #[inline]
+    fn add_assign(&mut self, rhs: Self) {
+        self.common.extend(rhs.common);
+        self.io.extend(rhs.io);
+        self.wasm.extend(rhs.wasm);
+    }
+}
+
+impl<T> FromIterator<Acc<T>> for Acc<Vec<T>> {
+    fn from_iter<A: IntoIterator<Item = Acc<T>>>(iter: A) -> Self {
+        iter.into_iter()
+            .fold(Acc::<Vec<T>>::default(), |mut acc, x| {
+                acc.common.push(x.common);
+                acc.io.push(x.io);
+                acc.wasm.push(x.wasm);
+                acc
+            })
+    }
+}
+
+impl<T> Acc<T> {
+    pub fn map<O>(self, mut mapper: impl FnMut(T) -> O) -> Acc<O> {
+        Acc {
+            common: mapper(self.common),
+            io: mapper(self.io),
+            wasm: mapper(self.wasm),
+        }
+    }
+    /// Assign this value to all non-common targets.
+    pub fn distribute(value: T) -> Self
+    where
+        T: Clone + Default,
+    {
+        Self {
+            common: T::default(),
+            io: value.clone(),
+            wasm: value,
+        }
+    }
+}
+
+impl<T: ToString> From<T> for Acc<Option<String>> {
+    #[inline]
+    fn from(common: T) -> Self {
+        Acc {
+            common: Some(common.to_string()),
+            ..Default::default()
+        }
+    }
+}
+
+impl<T> Acc<Vec<T>> {
+    /// Push to the common buffer.
+    #[inline]
+    pub fn push(&mut self, common: T) {
+        self.common.push(common)
+    }
+
+    /// Extend to the common buffer.
+    #[inline]
+    pub fn extend(&mut self, common: impl IntoIterator<Item = T>) {
+        self.common.extend(common)
+    }
+
+    #[inline]
+    pub fn push_acc(&mut self, acc: Acc<T>) {
+        let Acc { common, io, wasm } = acc;
+        self.common.push(common);
+        self.io.push(io);
+        self.wasm.push(wasm);
+    }
+
+    #[inline]
+    pub fn push_all(&mut self, item: T)
+    where
+        T: Clone,
+    {
+        self.common.push(item.clone());
+        self.io.push(item.clone());
+        self.wasm.push(item);
+    }
```

Comment:
Ah do we really need this case, since if sth is pushed into `common` I guess io and wasm should already auto have it

Reply:
Not really. The `common` buffer might introduce scope that is not present in the other buffers, so this is used to maintain code structure.